sources/core/Stride.Core.Reflection/TypeDescriptors/ObjectDescriptor.cs

```diff
@@ -237,6 +263,31 @@ where PrepareMember(member, metadataClassMemberInfos?.FirstOrDefault(x => x.Memb
             return memberList;
         }
 
+        static bool IsAccessibleThroughAccessModifiers(PropertyInfo property)
+        {
+            var get = property.GetMethod;
+            var set = property.SetMethod;
+
+            if (get == null)
+                return false;
+
+            bool forced = property.GetCustomAttribute<DataMemberAttribute>() is not null;
+
+            if (forced && (get.IsPublic || get.IsAssembly))
+                return true;
+
+            if (get.IsPublic)
+                return set?.IsPublic == true || set == null && TryGetBackingField(property, out _);
```

Comment:
This line is what requires attribute on `public { get; internal set; }` if get is public and set isn't we require attribute. Only if set is missing we check for backing fields.

Reply:
but internal setter is also valid?

Reply:
i dont think that a internal setter should be treated differently to a public setter , atleats in the future serializers

Reply:
Right, I wasn't very clear - I do know that this line is filtering out those kinds of properties. What I was wondering was why those are filtered out now, but weren't before.
Now, I believe that this is because the previous serializer is too lenient with properties that have a public getter.

Reply:
My guess would be that it's based in how reflection accessibility works on properties. In assembly processor we directly and only inspect getters and setters for accessibility. But here in the ObjectDescriptor we were looking for public properties. So a property can be public and have any accessibility modifier on its setter. We weren't checking it here. And later when checking if it's read only it would only check if a setter is present and not how it looks, because for reflection it didn't matter.